### PR TITLE
🐛 Some physical parameters are not passed correctly to simulators

### PR DESCRIPTION
**Pull request by Drewniok**
## Description

This PR fixes the problem that simulators always ran with the default values for ``epsilon_r`` and ``lambda_tf``. 

## Checklist:

- [x] The pull request only contains commits that are related to it.
- [x] I have added appropriate tests and documentation.
- [ ] I have made sure that all CI jobs on GitHub pass.
- [ ] The pull request introduces no new warnings and follows the project's style guidelines.

### CODE REVIEW

File: include/fiction/algorithms/simulation/sidb/sidb_simulation_parameters.hpp [hunk 1]
```diff
@@ -79,15 +72,26 @@ struct sidb_simulation_parameters
      * µ- is the energy transition level (0/-) (unit: eV).
      */
     double mu;
```
Comment (marcelwa) [inline include/fiction/algorithms/simulation/sidb/sidb_simulation_parameters.hpp:74]: While we're at it and are renaming stuff, can we fix this inconsistency with `mu` and `mu_p` as well and simply rename them to `mu_minus` and `mu_plus`?

Comment (Drewniok) [reply include/fiction/algorithms/simulation/sidb/sidb_simulation_parameters.hpp:74]: sure! I will change it quickly.

File: include/fiction/algorithms/simulation/sidb/sidb_simulation_parameters.hpp [hunk 2]
```diff
@@ -41,7 +41,7 @@ struct sidb_simulation_parameters
             lat_c{c},
             epsilon_r{relative_permittivity},
             lambda_tf{screening_distance},
-            mu{mu_minus},
+            mu_minus{mu_minus},
```
[the remark this answers is not in the document]
Comment (Drewniok) [reply include/fiction/algorithms/simulation/sidb/sidb_simulation_parameters.hpp:44]: Sorry, I was to quick at this point.

**Review by marcelwa — APPROVED** [review]

**Review by marcelwa — COMMENTED** [review]
Please consider CodeQL's comments.

**Comment by Drewniok** [comment]
> Please consider CodeQL's comments.

You are right, but I think it just does not know that we round the numbers before comparing them.

**Comment by marcelwa** [comment]
> > Please consider CodeQL's comments.
> 
> You are right, but I think it just does not know that we round the numbers before comparing them.

True, it most likely does not. However, it is considered a poor style to compare floating-point values by equality, since they can be imprecise very quickly. In this case, it does not matter whether you round them or not. If their value is large enough, there will be imprecision.

**Comment by Drewniok** [comment]
> > > Please consider CodeQL's comments.
> > 
> > 
> > You are right, but I think it just does not know that we round the numbers before comparing them.
> 
> True, it most likely does not. However, it is considered a poor style to compare floating-point values by equality, since they can be imprecise very quickly. In this case, it does not matter whether you round them or not. If their value is large enough, there will be imprecision.

I see. Thank you! I will change it.

**Review by marcelwa — APPROVED** [review]

**Pull request merged**